tasks/retrieval_deal.go

```diff
@@ -219,7 +219,35 @@ func (de *retrievalDealExecutor) executeAndMonitorDeal(ctx context.Context, upda
 				if event.Event == retrievalmarket.ClientEventDealAccepted {
 					stage = RetrievalStageDealAccepted
 					dealStage = RetrievalStages[stage]()
-					err := updateStage(ctx, stage, dealStage)
+					//there should be a deal now.
+					allRetrievals, err := de.node.ClientListRetrievals(ctx)
+					expectedPayload, parseErr := cid.Parse(de.task.PayloadCID)
+					if parseErr != nil {
+						return err
+					}
+					expectedID := abi.DealID(0)
+					if err == nil {
+						for _, maybeOurRetrieval := range allRetrievals {
+							if maybeOurRetrieval.PayloadCID.Equals(expectedPayload) {
+								//todo: maybe also match on && maybeOurRetrieval.Provider == peer.ID(de.task.Miner.x)
+								dealStage = AddLog(dealStage, fmt.Sprintf("DealID: %d", maybeOurRetrieval.ID))
+								expectedID = abi.DealID(maybeOurRetrieval.ID)
+								break
+							}
+						}
+					}
+					if expectedID != 0 {
+						allDeals, err := de.node.ClientListDeals(ctx)
+						if err != nil {
+							return err
+						}
+						for _, maybeOurDeal := range allDeals {
```

Comment:
This isn't going to work -- these are not the same DealID. The DealID on storage side is an onchain incrementing integer for all deals in the network. The DealID for retrieval is different -- it's just an off chain incrementing integer per client that has nothing to do with the storage deal id.

Reply:
There's not a really great correlation here -- but we could use payload CID + miner ID.... except that can definitely be duplicated. Our best piece of information is what we have locally when we create the retrieval from the storage deal -- when we create the retrieval deal after a storage, we should log the proposal CID.